presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -657,13 +675,95 @@ public ConnectorInsertTableHandle beginInsert(ConnectorSession session, Connecto
     {
         verifyJvmTimeZone();
 
-        throw new PrestoException(NOT_SUPPORTED, "INSERT not yet supported for Hive");
+        List<Boolean> partitionBitmap = null;
+
+        ImmutableList.Builder<String> columnNames = ImmutableList.builder();
+        ImmutableList.Builder<Type> columnTypes = ImmutableList.builder();
+
+        // call metastore to get table location, outputFormat, serde, partitions indices
+        Table table = null;
+        SchemaTableName tableSchemaName = getTableName(tableHandle);
+        try {
+            table = getMetastore().getTable(tableSchemaName.getSchemaName(), tableSchemaName.getTableName());
+        }
+        catch (NoSuchObjectException e) {
+            table = null;
+        }
+
+        checkNotNull(table, "Table %s does not exist", tableSchemaName.getTableName());
+        if (table.getSd().getNumBuckets() > 0) {
+            throw new UnsupportedOperationException("Insert not supported with Bucketed Tables");
+        }
+
+        String outputFormat = table.getSd().getOutputFormat();
+        SerDeInfo serdeInfo = table.getSd().getSerdeInfo();
+        String serdeLib = serdeInfo.getSerializationLib();
+        Map<String, String> serdeParameters = serdeInfo.getParameters();
```

Comment:
To keep this compatible with Hive you should pass both table and serde properties via InsertTableHandle to the HiveRecordSink. Then you should merge them into `properties` which you are passing to serde initialization method.

For instance when you create table in Hive using 

```
CREATE TABLE dummy(a int)
ROW FORMAT DELIMITED
   FIELDS TERMINATED BY '\007'
STORED AS TEXTFILE
TBLPROPERTIES ("serialization.null.format"="NULL");
```
Hive will serialize null values as a "NULL" string, while your implementation will still use default value.

Also it will be good to check that Hive has no other `properties` or something that should be merged into Serde initialization properties. You can check how it's done for Hive table reading in com.facebook.presto.hive.HiveSplitSourceProvider#getPartitionSchema